components/Forms/ComponentFactory.tsx

```diff
@@ -151,7 +158,7 @@ export const ComponentFactory: React.VFC<FactoryProps> = observer(
                         name={field.key}
                         label={field.label}
                         placeholder={field.placeholder ?? ''}
-                        onChange={field.handleChange}
+                        onChange={debounce(field.handleChange, 300)}
```

Comment:
I went with a 300ms debounce as it felt right, but let me know what you think. This solves the issue where you want to enter 65 as an age but when you hit 6 it already has an error on it